Mathlib/AlgebraicTopology/SimplicialSet/StrictSegal.lean

```diff
@@ -0,0 +1,156 @@
+/-
+Copyright (c) 2024 Emily Riehl. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mario Carneiro, Emily Riehl, Joël Riou
+-/
+import Mathlib.AlgebraicTopology.SimplicialSet.Nerve
+import Mathlib.AlgebraicTopology.SimplicialSet.Path
+import Mathlib.CategoryTheory.Functor.KanExtension.Adjunction
+import Mathlib.CategoryTheory.Functor.KanExtension.Basic
+
+/-!
+# Strict Segal simplicial sets
+
+A simplicial set `X` satisfies the `StrictSegal` condition if for all `n`, the map
+`X.spine n : X _[n] → X.Path n` is a bijection.
+
+Examples of `StrictSegal` simplicial sets are given by nerves of categories.
+
+TODO: Show that these are the only examples: that a `StrictSegal` simplicial set is isomorphic to
+the nerve of its homotopy category.
+
+`StrictSegal` simplicial sets have an important property of being 2-coskeletal which is proven
+elsewhere.
+-/
+
+universe v u
+
+open CategoryTheory
+
+open Simplicial SimplexCategory
+
+namespace SSet
+
+variable (X : SSet.{u})
+
+/-- A simplicial set `X` satisfies the strict Segal condition if its simplices are uniquely
+determined by their spine. -/
+class StrictSegal : Prop where
+  segal : ∀ (n : ℕ), Function.Bijective (X.spine n)
+
+variable {X} in
+/-- The diagonal of a simplex is the long edge of the simplex.-/
+def diagonal {n : ℕ} (Δ : X _[n]) : X _[1] := X.map ((mkOfDiag n).op) Δ
+
+namespace StrictSegal
+variable {X : SSet.{u}} [StrictSegal X] {n : ℕ}
+
+/-- In the presence of the strict Segal condition, a path of length `n` extends to an `n`-simplex
+whose spine is that path. -/
+noncomputable def spineToSimplex : Path X n → X _[n] :=
+  (Equiv.ofBijective _ (segal n)).invFun
```

Comment:
```suggestion
noncomputable def spineToSimplex : Path X n → X _[n] :=
  (Equiv.ofBijective _ (segal n)).symm
```
You could even record that this is bijective in the type, by making it an equiv. That preserves a bit more structure, and when needed proofs of inj/surj/bij will be readily available.

Reply:
Now this is part of the strict Segal structure.